quantlib/test/test_variance_swap.py

```diff
@@ -0,0 +1,190 @@
+import datetime
+import unittest
+
+import numpy as np
+
+from quantlib.settings import Settings
+from quantlib.instruments.option import EuropeanExercise, OptionType
+from quantlib.instruments.variance_swap import VarianceSwap, SwapType
+from quantlib.math.matrix import Matrix
+from quantlib.pricingengines.forward.replicating_variance_swap_engine import ReplicatingVarianceSwapEngine
+from quantlib.pricingengines.forward.mc_variance_swap_engine import MCVarianceSwapEngine
+from quantlib.processes.api import BlackScholesMertonProcess
+from quantlib.quotes import SimpleQuote
+from quantlib.termstructures.yields.flat_forward import FlatForward
+from quantlib.termstructures.volatility.equityfx.black_variance_curve import BlackVarianceCurve
+from quantlib.termstructures.volatility.equityfx.black_variance_surface import BlackVarianceSurface
+from quantlib.time.api import Date, today, NullCalendar, Actual365Fixed
+
+# Tests adapted from Quantlib test-suite/varianceswaps.cpp
+
+class VarianceSwapTestCase(unittest.TestCase):
+
+    def setUp(self):
+        """"""
+        self.today = today()
+        self.settings = Settings()
+        self.settings.evaluation_date = self.today
+        self.dc = Actual365Fixed()
+        self.spot = SimpleQuote(0.0)
+        self.q_rate = SimpleQuote(0.0)
+        self.q_ts = FlatForward(self.today, self.q_rate, self.dc)
+        self.r_rate = SimpleQuote(0.0)
+        self.r_ts = FlatForward(self.today, self.r_rate, self.dc)
+        self.values = {
+            'type' : SwapType.Long,
+            'strike' : 0.04,
+            'nominal' : 50000,
+            's' : 100.0,
+            'q' : 0.00,
+            'r' : 0.05,
+            't' : 0.246575,
+            'v' : 0.20,
+            'result' : 0.04189,
+            'tol' : 1.0e-4,
+        }
+        self.spot.value = self.values['s']
+        self.q_rate.value = self.values['q']
+        self.r_rate.value = self.values['r']
+        self.ex_date = self.today + int(self.values['t']*365+0.5)
+        #self.factory = (MakeMCVarianceSwapEngine(stochProcess).
```

Comment:
should this be part of another test? or just removed?

Reply:
removed in 1244cdc.  I had trouble getting the MCVarianceSwapEngine factory to work.  I'm not sure factory methods like these are really needed in Python. I can add this later if we want to include it.